lib/route/act.c

```diff
@@ -435,6 +442,43 @@ int rtnl_act_parse(struct rtnl_act **head, struct nlattr *tb)
 			tc->ce_mask |= TCA_ATTR_OPTS;
 		}
 
+		if (tb2[TCA_ACT_STATS]) {
+			struct nlattr *tb3[TCA_STATS_MAX + 1];
+
+			err = nla_parse_nested(tb3, TCA_STATS_MAX, tb2[TCA_ACT_STATS],
+					       tc_act_stats_policy);
+			if (err < 0)
+				return err;
+
+			if (tb3[TCA_STATS_BASIC]) {
+				struct gnet_stats_basic *bs;
+
+				bs = nla_data(tb3[TCA_STATS_BASIC]);
+				tc->tc_stats[RTNL_TC_BYTES] = bs->bytes;
```

Comment:
I think the netlink attribute is only guaranteed to have an alignment of 4 (`sizeof(uint32_t)`). Maybe (just maybe), we also assume that `sizeof(int) == 4`, so it would also be valid for `int`/`unsigned` fields (but I think we should not assume that).

However, `bs->bytes` and `struct gnet_stats_rate_est64` are uint64. We need to memcpy them.

Maybe a simpler solution is to
```C
    struct gnet_stats_basic bs;

    memcpy(&bs, nla_data(tb3[TCA_STATS_BASIC]), sizeof(bs));
    tc->tc_stats[RTNL_TC_BYTES] = bs.bytes;
```
then we don't need to reason about each field individually, which type they are and when/how to memcpy.

Reply:
for this specific attribute, nla_data is 4 bytes alignment and payload is padded to multiple of 8 bytes. If I well get the problem, it's about unaligned access when doing cast + assignment inside the netlink payload. I see this is done elsewhere in libnl3 - for [qdisc stats](https://github.com/thom311/libnl/blob/main/lib/route/tc.c#L110) - but yes,  I can change it (though I'm unsure I understood what kind of problem it can solve).

Thanks for reviewing!

Reply:
the problem to avoid, is unaligned memory access (in this case, reading a uint64 integer).

As netlink attributes have a guaranteed alignment of "only" 4 bytes, I think it can happen that  `nla_data(tb3[TCA_STATS_BASIC])` starts at an address with `addresss%8==4`. Or why not?

I don't understand why that would *not* be a problem. I think https://github.com/thom311/libnl/blob/main/lib/route/tc.c#L110 is a bug. Or why not?

Reply:
... understood. Done, just force-pushed a v2 where I rebased and converted parser of `TCA_STATS_BASIC` and `TCA_STATS_RATE_EST64` to copy on a local struct allocated on the stack. The other 2 stats, namely `TCA_STATS_QUEUE` and `TCA_STATS_RATE_EST` have only 32-bit wide members, so all accessed should already be aligned to 4-bytes.